### PR TITLE
Improve query params for enabling strict/dev mode

### PR DESCRIPTION
**Pull request by ecraig12345**
Quick follow-up for #14327: right after merging that, I realized it doesn't make sense to require people to type *both* `?strict=all&isDev=1`, and the inconsistent naming was annoying. So...
- Rename `isDev` to `dev`
- Make `strict` imply `dev` (`React.StrictMode` only works in React dev builds)

### CODE REVIEW

File: apps/fabric-website/index.html [hunk 1]
```diff
@@ -156,7 +156,7 @@
     <script type="text/javascript">
       // @ts-check
       var isLocal = location.hostname === 'localhost' || location.hostname.indexOf('ngrok.io') > -1;
-      var isDev = isLocal || getParameterByName('isDev') === '1';
+      var isDev = isLocal || getParameterByName('dev') === '1' || !!getParameterByName('strict');
```
Comment (xugao) [inline apps/fabric-website/index.html:159]: where is `var isDev` being used?

Comment (xugao) [reply apps/fabric-website/index.html:159]: also, looks odd to have both `isDev` and `isProduction`.

Comment (ecraig12345) [reply apps/fabric-website/index.html:159]: Oops, there's some logic I forgot to copy from homepage.htm.

Comment (ecraig12345) [reply apps/fabric-website/index.html:159]: `isDev` determines whether to use dev mode (non-minified scripts and dev mode react). `isProduction` determines whether to use the production manifest or dogfood/staging manifest.

**Pull request merged**